src/python/WMCore/MicroService/MSPileup/MSPileup.py

```diff
@@ -98,12 +99,27 @@ def updatePileup(self, pdict):
         """
         MSPileup update API to update corresponding pileup document in data layer.
 
-        :param pdict: input MSPilup data dictionary
+        :param pdict: input MSPilup data dictionary or partial pileup spec
+        {"containerFraction": int, "pileupName": string}
         :return: results of MSPileup data layer (list of dicts)
         """
         self.authMgr.authorizeApiAccess('ms-pileup', 'update')
+        keys = sorted(pdict.keys())
+        if keys == ['containerFraction', 'pileupName']:
```

Comment:
Why it is different in different python versions? Please provide an example, list is a basic data type and never change. Said that, we can't check it with `if "containerFraction" in pdict:` either since `containerFraction` is present in both pileup document and partial pileup spec. Unless you provide better argument here I'm leaving code as is.